woocommerce/woocommerce-android
# Issue/1482 step2 magic link signin changes

Partially fixes #1482. This PR completes the second step of the Jetpack sign in with self credentials as defined in the master thread.

### Changes
- Modified UI in `LoginMagicLinkRequestFragment` and `LoginMagicLinkSentFragment` according to new design.
- Created a new class: `MagicLinkInterceptFragment` that handles the magic link login process. This was previously handled by the `MainActivity`. This logic has been migrated to the new class. 

### Behaviour
(LEFT: Before . RIGHT: After)  
<img width="300" src="https://user-images.githubusercontent.com/22608780/67493103-01aa2a80-f695-11e9-84f1-6b944cbad3c9.gif"> . <img width="300" src="https://user-images.githubusercontent.com/22608780/67492935-cd366e80-f694-11e9-9108-3077c4b56e99.gif">

### Notes:
- The are TODO's added to the PR to add events to the button clicks. This will take place in another PR.
- Note that this PR is to be merged to a feature branch.
- I've kept the option to enter the password as well since the user might not always have the email connected to their account on their phone. This is not part of the design flow though. Let me know if there are any objections and we can certainly hide it.

### Testing:
- Verify that the current flow with magic link is working correctly.
- Verify the current flow to enter password instead of the magic flow link, is working correctly.


Update release notes:
